### ISSUE TITLE
oneOf is not treated the same as anyOf

### ISSUE THREAD

**Issue by ofek** [koxudaxi/datamodel-code-generator#333]
**Describe the bug**

The docs indicate that they are both treated as `typing.Union`

**To Reproduce**

Example schema:

```yaml
components:
  schemas:
    Config:
      properties:
        setting:
          type: object
          additionalProperties:
            oneOf:  # anyOf
            - type: string
            - type: array
              items:
                type: string
```

Used commandline:

```
$ datamodel-codegen --input-file-type openapi --strip-default-none --input test.yaml --output model.py
```

`oneOf`:

```python
from __future__ import annotations

from typing import Any, Dict, Optional

from pydantic import BaseModel, Extra


class Setting(BaseModel):
    class Config:
        extra = Extra.allow

    __root__: Any


class Config(BaseModel):
    setting: Optional[Dict[str, Setting]]
```

`anyOf`:

```python
from __future__ import annotations

from typing import Dict, List, Optional, Union

from pydantic import BaseModel, Extra


class Setting(BaseModel):
    class Config:
        extra = Extra.allow

    __root__: Union[str, List[str]]


class Config(BaseModel):
    setting: Optional[Dict[str, Setting]]
```

**Version:**
 - OS: Windows 10
 - Python version: 3.8
 - datamodel-code-generator version: 0.7.2

**Comment by koxudaxi** [comment]
@ofek 
Good point!!
I give up the feature of `oneOf`.
https://github.com/koxudaxi/datamodel-code-generator/pull/147

I will fix to treat `oneOf` strictly.
I may add an option for this feature to keep current behavior.

Thank you!!

**Comment by ofek** [comment]
Ah, I was expecting the same behavior as `anyOf`, so I'm just using that now

**Comment by koxudaxi** [comment]
I'm sorry, I have misread the problem.
`oneOf` may be broken 🤔 
I will check it.

**Comment by koxudaxi** [comment]
@ofek 
I have released a fixed version `0.7.3`

**Comment by ofek** [comment]
Fixed!!! 🙇
